generate_yaml/entrypoint.sh

```diff
@@ -87,6 +87,9 @@ main(){
       launchdarkly_sdkkey=$INPUT_LAUNCHDARKLY_SDKKEY_STAGING
       post_action_events_pub_sub_project_id=$INPUT_POST_ACTION_EVENTS_PUBSUB_PROJECT_ID_STAGING
       post_action_events_pub_sub_credentials_json=$INPUT_POST_ACTION_EVENTS_PUBSUB_CREDENTIALS_JSON_STAGING
+      raw_data_pub_sub_topic=$INPUT_RAW_DATA_PUBSUB_TOPIC_STAGING
```

Comment:
1. Is `raw_data_pub_sub_topic` needed? I cannot see in this PR where is used.
2. Looking at the set value, shouldn't these variables have the `staging` suffix?

Reply:
1. You're absolutely right, I probably thought `raw_data_pub_sub_topic` was needed but it's not. So, I'll remove that line after commenting here. Thanks for catching that.

2. That's also a good question, and my reply is: *I don't think (and I'm sure) they should*.

    I know it's confusing, so here's my attempt to explain:
    I'm going to use the `post_action_events_pub_sub_project_id` variable as an example:

    When we're deploying to staging, which is defined [here](https://github.com/loyalguru/actions/blob/5a634bc3b000c6654cd8a13633fbaf7e82b3d354/generate_yaml/entrypoint.sh#L34-L39) and later "checked" [here](https://github.com/loyalguru/actions/blob/5a634bc3b000c6654cd8a13633fbaf7e82b3d354/generate_yaml/entrypoint.sh#L83)

    The [value](https://github.com/loyalguru/actions/blob/5a634bc3b000c6654cd8a13633fbaf7e82b3d354/generate_yaml/entrypoint.sh#L88) of `post_action_events_pub_sub_project_id` will be the value of `$INPUT_POST_ACTION_EVENTS_PUBSUB_PROJECT_ID_STAGING`, but when we're deploying to production, the [value](https://github.com/loyalguru/actions/blob/5a634bc3b000c6654cd8a13633fbaf7e82b3d354/generate_yaml/entrypoint.sh#L78) of `post_action_events_pub_sub_project_id` will be the value of `$INPUT_POST_ACTION_EVENTS_PUBSUB_PROJECT_ID`.

    So the same variable (in the shell script) has a different value depending on the environment, and is then [used](https://github.com/loyalguru/actions/blob/5a634bc3b000c6654cd8a13633fbaf7e82b3d354/generate_yaml/entrypoint.sh#L126-L128) to "generate" the `.yaml` file.
    
    Does that make more sense?

Reply:
@josesolanaLoyalGuru there was a typo in my last comment, sorry for that.

I wrote:
> *I don't think (and I'm sure) they should.*

But I meant:
> *I don't think (and I'm sure) they shouldn't.*

I edited it since then but if you read it via email you must have been confused.

Reply:
Oh, I get it! Thank you for the explanation. I had forgotten how it worked.

Reply:
No worries